File: sieve_of_eratosthenes/sieveOfEratosthenes.py
def sieveOfEratosthenes(n):
    if n < 2:
        return []

    bools = [False, False]

    for _ in range(2, n + 1):
        bools.append(True)

    for i in range(2, round(n ** 0.5)):
        if bools[i] == True:
            j = i * i
            x = 1
            if j <= n:
                bools[j] = False
            while j <= n:
                j = i * i + x * i
                if j <= n:
                    bools[j] = False
                x += 1

    primes = []
    for i in range(2, n + 1):
        if bools[i] == True:
            primes.append(i)

    return primes

print(sieveOfEratosthenes(100))

File: sieve_of_eratosthenes/sieveOfEratosthenes.js
var sieveOfEratosthenes = (n) => {
  let bools = [];

  for (let i = 2; i <= n; i++) bools.push(true);

  for (let i = 2; i < Math.sqrt(n); i++) {
  	// For index values which remain true in the set of 'bools'
  	// after the while loops (below) -- those are primes.
  	if (bools[i] === true) {
  		// 'i' is a prime number.
  		let j = i*i,
  		    x = 1;
  		bools[j] = false;
  		while (j <= n) {
  			// Set every multiple of 'i' to false.
  			j = i * i + x * i;
        bools[j] = false;
        x++;
  		}
  	}
  } 
  let primes = [];
  for (let i = 2; i <= n; i++) {
  	if (bools[i] === true) primes.push(i);
  }

  return primes;
};